codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -214,9 +218,41 @@ public ChannelFuture sendFlowControlledFrame(ChannelHandlerContext ctx, Http2Str
     }
 
     @Override
-    public ChannelFuture lastFlowControlledFrameSent(Http2Stream stream) {
-        FlowState state = state(stream);
-        return state != null ? state.lastNewFrame() : null;
+    public ChannelFuture sendHeaders(ChannelHandlerContext ctx, Http2Stream stream, Http2Headers headers,
+            int streamDependency, short weight, boolean exclusive, int padding,
+            boolean endOfStream, ChannelPromise promise) {
+      boolean isFirst = !sentFrame;
```

Comment:
Thanks for the clarification.  I think it would be nice if the verification could stay in `DefaultHttp2ConnectionEncoder`, but if that complicates things (or is not seeing the whole picture) then lets reassess.  I could see how if we are queuing things the state could change by the time we actually try to send the frame (data or headers)....haven't thought about it too much but this may be an issue.